xcdat/axis.py

```diff
@@ -0,0 +1,204 @@
+from typing import List, Literal, TypedDict
+
+import numpy as np
+import xarray as xr
+
+
+@xr.register_dataset_accessor("axis")
+class AxisAccessor:
+    """A class used to represent an AxisAccessor (xarray dataset accessor)
+
+    :param xarray_obj: The Dataset object to be extended
+    :type xarray_obj: xr.DataArray
+    """
+
+    Axis = Literal["lat", "lon"]
+    AxesMapValue = TypedDict(
+        "AxesMapValue", {"coords": List[str], "bounds_vars": List[str]}
+    )
+    AxesMap = TypedDict("AxesMap", {"lat": AxesMapValue, "lon": AxesMapValue})
+
+    # Mapping of generic axes names to the possible names for Dataset
+    # coordinates and boundary variables
+    axes_map: AxesMap = {
+        "lat": {
+            "coords": ["lat", "latitude"],
+            "bounds_vars": ["lat_bnds", "latitude_bnds"],
+        },
+        "lon": {
+            "coords": ["lon", "longitude"],
+            "bounds_vars": ["lon_bnds", "longitude_bnds"],
+        },
+    }
+
+    def __init__(self, xarray_obj: xr.Dataset):
+        self._obj = xarray_obj
+
+    def get_bounds(self, axis: Axis) -> xr.DataArray:
+        """Get bounds for an axis.
+
+        :param axis: "lat" or "lon" axis
+        :type axis: Axis
+        :return: The axis bounds, either existing bounds or calculated bounds
+        :rtype: xr.DataArray
+        """
+        bounds_vars = AxisAccessor.axes_map[axis]["bounds_vars"]
+        matching_bounds_var = None
+
+        for var in bounds_vars:
+            matching_bounds_var = self._obj.data_vars.get(var)
+            if matching_bounds_var is not None:
+                break
+
+        if matching_bounds_var is None:
+            return self._generate_bounds(axis)
+
+        return matching_bounds_var
+
+    def _generate_bounds(self, axis: Axis, width: float = 1) -> xr.DataArray:
```

Comment:
@chengzhuzhang @pochedls This method and its children methods are based on CDAT's [`genGenericBounds` ](https://github.com/CDAT/cdms/blob/dd41a8dd3b5bac10a4bfdf6e56f6465e11efc51d/Lib/axis.py#L2171) method.

If you do a comparison, you'll see some resemblance in the code. However, I've broken up the function into modular methods, and most of the conditional statements have been refactored for readability.